File: Trunk/vpPriV100GrupoMundifios/DyStar/Compras/EditorCompras/CmpIsEditorCompras.cs
using Generico;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs;
using Primavera.Extensibility.Purchases.Editors;
using static BasBE100.BasBETiposGcp;

namespace DyStar
{
    public class CmpIsEditorCompras : EditorCompras
    {
        public override void AntesDeGravar(ref bool Cancel, ExtensibilityEventArgs e)
        {
            base.AntesDeGravar(ref Cancel, e);

            if (Module1.VerificaToken("DyStar") == 1)
            {
                // #######################################################################################################################################################
                // #Adiciona na ultima linha de artigos um comentário se os fornecedore forem DyStar ou TCC. Bruno - 08/08/2019, tem de ser no fim de todas as condições #
                // #######################################################################################################################################################

                if (((this.DocumentoCompra.Entidade == "1809") | (this.DocumentoCompra.Entidade == "1812")) & (this.DocumentoCompra.Tipodoc == "ECF"))
                {
                    if (!BSO.Compras.Documentos.Existe(this.DocumentoCompra.Filial, this.DocumentoCompra.Tipodoc, this.DocumentoCompra.Serie, this.DocumentoCompra.NumDoc))
                        BSO.Compras.Documentos.AdicionaLinhaEspecial(this.DocumentoCompra, compTipoLinhaEspecial.compLinha_Comentario, 0, "Os artigos incluídos nesta ECF devem ser entregues na sua embalagem original, com o rótulo original, incluindo o nome do produto, o nome do fabricante/distribuidor e número do lote do produto químico.");
                }
            }
        }
    }
}

File: Branches/VBInicial/vpPriV100GrupoMundifios/DyStar/Compras/EditorCompras/CmpIsEditorCompras.vb
Imports Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs
Imports Primavera.Extensibility.Purchases.Editors
Imports Primavera.Extensibility.Attributes
Imports Vimaponto.PrimaveraV100.Clientes.GrupoMundifios.Generico
Imports StdBE100
Imports BasBE100.BasBETiposGcp

Namespace DyStar
    Public Class CmpIsEditorCompras
        Inherits EditorCompras

        Public Overrides Sub AntesDeGravar(ByRef Cancel As Boolean, e As ExtensibilityEventArgs)
            MyBase.AntesDeGravar(Cancel, e)

            If Module1.VerificaToken("DyStar") = 1 Then

                '#######################################################################################################################################################
                '#Adiciona na ultima linha de artigos um comentário se os fornecedore forem DyStar ou TCC. Bruno - 08/08/2019, tem de ser no fim de todas as condições #
                '#######################################################################################################################################################

                If ((Me.DocumentoCompra.Entidade = "1809") Or (Me.DocumentoCompra.Entidade = "1812")) And (Me.DocumentoCompra.Tipodoc = "ECF") Then
                    If Not BSO.Compras.Documentos.Existe(Me.DocumentoCompra.Filial, Me.DocumentoCompra.Tipodoc, Me.DocumentoCompra.Serie, Me.DocumentoCompra.NumDoc) Then
                        BSO.Compras.Documentos.AdicionaLinhaEspecial(Me.DocumentoCompra, vdTipoLinhaEspecial.vdLinha_Comentario, 0, "Os artigos incluídos nesta ECF devem ser entregues na sua embalagem original, com o rótulo original, incluindo o nome do produto, o nome do fabricante/distribuidor e número do lote do produto químico.")
                    End If
                End If

                '#######################################################################################################################################################
                '#Adiciona na ultima linha de artigos um comentário se os fornecedore forem DyStar ou TCC. Bruno - 08/08/2019, tem de ser no fim de todas as condições #
                '#######################################################################################################################################################

            End If

        End Sub

    End Class
End Namespace
